src/commands/api/update.js

```diff
@@ -11,6 +11,20 @@ class UpdateAPICommand extends BaseCommand {
   
   async updateApi(owner, name, version, flags) {
     const isPrivate = flags.visibility === 'private'
+
+    if (!flags.file && flags.visibility) {
+      const updateApiObj = {
+        pathParams: [owner, name, version, 'settings', 'private'],
+        body: JSON.stringify({ private: isPrivate })
+      }
+  
+      return await this.executeHttp({
+          execute: () => putApi(updateApiObj), 
+          onResolve: this.logCommandSuccess({ owner, name, version }),
+          options: { resolveStatus: [403] }
+      })
+    }
+
     const updateApiObj = {
```

Comment:
Are we still executing the updateApi after setting the API to private/public? Does this still overwrite the API?

Reply:
Yes in case --visibility is not provided or --file is there